src/Numerics/DGMethods/DGModel_kernels.jl

```diff
@@ -272,7 +268,7 @@ fluxes, respectively.
                 @unroll for n in 1:Nqk
                     MI = local_MI[n]
                     @unroll for s in 1:num_state_prognostic
-                        local_tendency[n, s] += MI * s_D[k, n] * local_flux_3[s]
+                        local_tendency[n, s] += MI * D[k, n] * local_flux_3[s]
```

Comment:
Did anyone look at the performance difference of remove `D` from shared memory?

Reply:
So @mwarusz and I plan to address this in a separate PR. The main issue here is that we weren't sure how to best load this matrix into shared memory given that the thread extents (`Nq[1], Nq[2], Nq[3]`) are now all different. Maciej and I ran into bounds errors in the kernels when we tried to vary the polynomial order without this change. We think this needs to be fixed, but it requires some rewriting of these kernels which is beyond the scope of this initial PR. I think it's a good thing to bring up though. @mwarusz Should we create an issue?

Reply:
Sounds good.

Reply:
Can we use max(Nq) for sizing D in Shmem? This is what I did for strain rate tensor computations in diagnostics.